addon/components/line-chart/component.js

```diff
@@ -186,9 +186,11 @@ export default BaseChartComponent.extend({
         if (this.isIntervalInRange(xTimeScale, indicatorDate)) {
             let currentTick = d3.select('.line-chart > svg > g > g.axis').selectAll('g.tick')
                 .filter(d => d.toString() === indicatorDate.toString());
-            if (currentTick.select('text').text().indexOf('\u25C6') === -1) {
-                let tickHtml = this.isIntervalIncluded(xTimeScale.ticks(this.get('xAxis').ticks), indicatorDate) ? `\u25C6 ${currentTick.text()}` : '\u25C6';
-                currentTick.select('text').html(tickHtml);
+            if (currentTick && !currentTick.empty()) {
+                if (currentTick.select('text').text().indexOf('\u25C6') === -1) {
```

Comment:
Could combine the two `if` statements to one line if you'd like

Reply:
The error comes from `currentTick.select('text').text()` in the second `if` statement, though. We can't do that check until we know that `currentTick.select('text')` is not null.
...Or does it not evaluate all the statements in the `if` if the first one is false?

Reply:
Correct, JavaScript has short-circuit execution of logical operators, so you can do things like `const x = thing && thing.value;` and not have to worry about getting a property access on `undefined` error on the dot operator in `thing.value` because it will only execute if `thing` is truthy.

Reply:
Got it, thanks!